src/echo/echo.rs

```diff
@@ -36,119 +36,93 @@ static HELP: &str = r#"
  \\xHH    byte with hexadecimal value HH (1 to 2 digits)
 "#;
 
-enum Base {
-    B8 = 8,
-    B16 = 16,
-}
-
-struct Opts {
-    newline: bool,
-    escape: bool,
+fn parse_code(input: &mut Peekable<Chars>, base: u32, max_digits: u32, bits_per_digit: u32) -> Option<char> {
+    let mut ret = 0x80000000;
+    for _ in 0..max_digits {
+        match input.peek() {
```

Comment:
```suggestion
        match input.peek().and_then(|c| c.to_digit(base)) {
```

so that you don't need to match again on the line below.

Reply:
Actually, why is this a peek instead of a next?

Reply:
Good suggestion! Done.